src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newStr = '';
+
+    for (const i of this) {
+      if (i != null) {
```

Comment:
it's important to use strict equality. loose equality can behave in strange ways sometimes

Reply:
corrected